clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3569,6 +3557,69 @@ protected enum DelegateConsumerRecordResult {
     SKIPPED_MESSAGE
   }
 
+  private void recordNearlineLocalBrokerToReadyToServerLatency(
+      String storeName,
+      int versionNumber,
+      PartitionConsumptionState partitionConsumptionState,
+      KafkaMessageEnvelope kafkaMessageEnvelope,
+      LeaderProducedRecordContext leaderProducedRecordContext) {
+    long afterProcessingRecordTimestampMs = System.currentTimeMillis();
+    if (!isUserSystemStore() && isHybridMode() && partitionConsumptionState.hasLagCaughtUp()
```

Comment:
Haven't checked the logic in details yet but I think we can make line 3567 - 3579 a function, that's to determine whether we need to record the latency, so it improves the readability. Some javadoc would be helpful too

Reply:
Shortened the condition . since it fits in 1 line don't think adding a method will increase readability. added javadoc as well